### PR TITLE
[Compression] fix typehints

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

File: nni/algorithms/compression/v2/pytorch/base/compressor.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import collections
 import logging
-from typing import List, Dict, Optional, Tuple, Any
+from typing import Any, List, Dict, OrderedDict, Optional, Tuple
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/base/compressor.py:6]: How is `OrderedDict` get used? In Python 3.7+ ordinary `dict` has guaranteed order. I think `OrderedDict` is not very useful in most cases now.

File: nni/algorithms/compression/v2/pytorch/base/compressor.py [hunk 2]
```diff
@@ -182,7 +207,7 @@ def set_wrappers_attribute(self, name: str, value: Any):
         value
             Value of the variable.
         """
-        for wrapper in self.get_modules_wrapper():
+        for _, wrapper in self.get_modules_wrapper().items():
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/base/compressor.py:210]: I think you mean `self.get_modules_wrapper().values()`.

File: nni/algorithms/compression/v2/pytorch/pruning/tools/task_generator.py [hunk 3]
```diff
@@ -327,11 +327,12 @@ def init_pending_tasks(self) -> List[Task]:
 
     def generate_tasks(self, task_result: TaskResult) -> List[Task]:
         # initial/update temp config list
-        if self._temp_config_list is None:
+        if len(self._temp_config_list) == 0:
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/pruning/tools/task_generator.py:330]: Python's official style guide suggests to use `if not self._temp_config_list:`. But this only applies to Python `list`, not numpy ndarray.

Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/pruning/tools/task_generator.py:330]: got it

File: nni/algorithms/compression/v2/pytorch/utils/pruning.py [hunk 4]
```diff
@@ -229,7 +229,7 @@ def compute_sparsity(origin_model: Module, compact_model: Module, compact_model_
     return current2origin_sparsity, compact2origin_sparsity, mask2compact_sparsity
 
 
-def get_model_weights_numel(model: Module, config_list: List[Dict], masks: Dict[str, Dict[str, Tensor]] = {}) -> Dict:
+def get_model_weights_numel(model: Module, config_list: List[Dict], masks: Dict[str, Dict[str, Tensor]] = {}) -> Tuple[Dict, Dict]:
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/utils/pruning.py:232]: If this is a public API, it should clarify the key type and value type of returned dict.

File: nni/algorithms/compression/v2/pytorch/base/compressor.py (hunk 1 above)
Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/base/compressor.py:6]: just want a guaranteed order, I will change it to `dict`.

**Review by ultmaster — APPROVED** [review]

File: nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py [hunk 5]
```diff
@@ -46,7 +46,9 @@ def _iterative_pruner_reset(self, model: Module, config_list: List[Dict] = [], m
     def allocate_sparsity(self, new_config_list: List[Dict], model: Module, masks: Dict[str, Dict[str, Tensor]]):
         self._iterative_pruner_reset(model, new_config_list, masks)
         self.iterative_pruner.compress()
-        _, _, _, _, config_list = self.iterative_pruner.get_best_result()
+        best_result = self.iterative_pruner.get_best_result()
+        assert best_result is not None, 'Best result does not exist, iterative pruner may not start pruning.'
+        _, _, _, _, config_list = best_result
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py:51]: Irrelevant to this PR, but this looks really strange to me...

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py [hunk 6]
```diff
@@ -654,11 +656,11 @@ def _validate_config_before_canonical(self, model: Module, config_list: List[Dic
...
     def _choose_activation(self, activation: str = 'relu') -> Callable:
         if activation == 'relu':
-            return nn.functional.relu
+            return F.relu
         elif activation == 'relu6':
-            return nn.functional.relu6
+            return F.relu6
         else:
-            raise 'Unsupported activatoin {}'.format(activation)
+            raise Exception('Unsupported activatoin {}'.format(activation))
```
Comment (liuzhe-lz) [inline nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py:663]: Should use a more concrete exception type. (`RuntimeError`, `ValueError`, etc)

**Review by liuzhe-lz — APPROVED** [review]

File: nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py (hunk 5 above)
Comment (J-shang) [reply nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py:51]: agree, will change these in future

**Pull request merged**